tockloader/openocd.py

```diff
@@ -408,16 +408,23 @@ def run_terminal(self):
             cleanup.append(ocd_p.wait)
             cleanup.append(ocd_p.kill)
 
-            # Delay to give the connection time to start before running
-            # the RTT listener.
-            time.sleep(1)
-            if ocd_p.poll():
-                return
-
-            logging.status("Listening for messages.")
             listener = socket.socket()
-            listener.connect(("127.0.0.1", 9999))
+            for i in reversed(range(3)):
```

Comment:
```suggestion
            const MAX_TRIES = 3;
            for i in 0..MAX_TRIES:
```

This is just personal preference, but I think it is a bit less confusing